packages/main/src/Select.ts

```diff
@@ -1124,7 +916,7 @@ class Select extends UI5Element implements IFormElement {
 	}
 
 	get selectedOptionIcon() {
-		return this.selectedOption && this.selectedOption.icon;
+		return this.selectedOption && (this.selectedOption as Option).icon;
```

Comment:
we can also do it like: `return this.selectedOption && "icon" in this.selectedOption && this.selectedOption.icon;` instead of casting. but it will do the same basically.